src/components/analyses/RenameAnalysisDialog.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * A dialog that allows users to rename an analysis.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import { Field, Form, Formik } from "formik";
+import { useTranslation } from "i18n";
+
+import { validateDiskResourceName } from "../data/utils";
+
+import ids from "./ids";
+import UtilIds from "../utils/ids.js";
+
+import { build, FormTextField } from "@cyverse-de/ui-lib";
+
+import {
+    Button,
+    CircularProgress,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    InputAdornment,
+} from "@material-ui/core";
+
+function RenameAnalysisDialog(props) {
+    const { open, isLoading, onClose, handleRename, selectedAnalysis } = props;
+
+    // Using separate hooks so that analyses and data message keys
+    // don't require a prefix.
+    const { t } = useTranslation(["analyses", "common"]);
+    const { t: dataI18N } = useTranslation("data");
```

Comment:
I have similar issue in my code. I can't use array syntax as It would only pick the first namespace in the array 😕. Did you encounter any issue as well ?

Reply:
Yep, unfortunately, according to the documentation it will only look up the key in the first namespace of the array if the key doesn't have a prefix:
https://react.i18next.com/latest/usetranslation-hook#usetranslation-params

That's why below I have to use `t("common:cancel")` and `t("common:ok")`. The messages used in `validateDiskResourceName` can't use the `data:` namespace yet, since apps is still using `getMessage` for now, but once that's converted with `useTranslation`, then I think the `data` namespace can be used in that function and also included in this array.